.github/actions/deploy-cluster-run-tests/action.yml

```diff
@@ -0,0 +1,20 @@
+name: "Deploy Kind cluster and run tests"
+description: "Loads docker image previously built, installs kind, builds and runs e2e tests and exports logs and reports"
+runs:
+  using: "composite"
+  steps:
+  - name: Load docker image
+    run: |
+      docker load --input image.tar
+    shell: bash
```

Comment:
this steps depends that the image has be download in the current path, I'd keep it with the same code that downloads the image in the main workflow, or is easy it can go out of sync, i.e. imaging we pass the image as a cache object instead of an artificats, this step will fail. 
However, the next two steps sounds good, they will always work